cpp/src/arrow/ipc/reader.cc

```diff
@@ -936,6 +962,104 @@ Status MakeSparseTensorWithSparseCSRIndex(
 
 }  // namespace
 
+namespace internal {
+
+namespace {
+
+Status GetSparseTensorBodyBufferCount(SparseTensorFormat::type format_id,
+                                      size_t* buffer_count) {
+  switch (format_id) {
+    case SparseTensorFormat::COO:
+      *buffer_count = 2;
+      break;
+
+    case SparseTensorFormat::CSR:
+      *buffer_count = 3;
+      break;
+
+    default:
+      return Status::Invalid("Unrecognized sparse tensor format");
+  }
+
+  return Status::OK();
+}
+
+Status CheckSparseTensorBodyBufferCount(
+    const IpcPayload& payload, SparseTensorFormat::type sparse_tensor_format_id) {
+  size_t expected_body_buffer_count;
+
+  RETURN_NOT_OK(GetSparseTensorBodyBufferCount(sparse_tensor_format_id,
+                                               &expected_body_buffer_count));
+  if (payload.body_buffers.size() != expected_body_buffer_count) {
+    return Status::Invalid("Invalid body buffer count for a sparse tensor");
+  }
+
+  return Status::OK();
+}
+
+}  // namespace
+
+Status ReadSparseTensorBodyBufferCount(const Buffer& metadata, size_t* buffer_count) {
+  SparseTensorFormat::type format_id;
+
+  RETURN_NOT_OK(internal::GetSparseTensorMetadata(metadata, nullptr, nullptr, nullptr,
+                                                  nullptr, &format_id));
+  return GetSparseTensorBodyBufferCount(format_id, buffer_count);
+}
+
+Status ReadSparseTensorPayload(const IpcPayload& payload,
```

Comment:
This is almost the same code as `ReadSparseTensor` below. Instead of copy/pasting, can you refactor `ReadSparseTensor` around `ReadSparseTensorPayload`?

Reply:
Also applies to the `ReadSparseXXXIndex` below.

Reply:
@rok Let me perform the refactoring of these functions.

Reply:
@mrkn that would be great. Note I've pushed some changes in python.

Reply:
@rok The refactoring has been done.